packages/builder/src/extrinsic/ExtrinsicBuilder.utils.ts

```diff
@@ -61,3 +62,41 @@ export function getExtrinsicAccount(address: string) {
         },
       };
 }
+
+export function isXcmV4(xcmVersion: XcmVersion): boolean {
+  return xcmVersion >= XcmVersion.v4;
+}
+
+export function normalizeX1(
+  xcmVersion: XcmVersion,
+  versionedObject: Record<string, AnyJson>,
+) {
+  if (!isXcmV4(xcmVersion)) {
+    return versionedObject;
+  }
+  const normalizedObject = { ...versionedObject };
+  const interior = normalizedObject.interior;
+  if (interior && typeof interior === 'object' && !Array.isArray(interior)) {
+    const key = 'X1' in interior ? 'X1' : 'x1' in interior ? 'x1' : null;
+    if (key && !Array.isArray(interior[key])) {
+      interior[key] = [interior[key]];
+    }
+  }
+
+  return normalizedObject;
+}
```

Comment:
Suggestion to make it a bit easier to read.

```suggestion
export function normalizeX1(xcmVersion: XcmVersion, versionedObject: Record<string, AnyJson>) {
  if (!isXcmV4(xcmVersion)) return versionedObject;

  const normalizedAssetType = { ...versionedObject };
  const interior = normalizedAssetType.interior;

  if (interior?.X1 && !Array.isArray(interior.X1)) {
    interior.X1 = [interior.X1];
  } else if (interior?.x1 && !Array.isArray(interior.x1)) {
    interior.x1 = [interior.x1];
  }

  return normalizedAssetType;
}
```